nx_parallel/algorithms/tournament.py

```diff
@@ -25,33 +27,50 @@ def is_reachable(G, s, t, get_chunks="chunks"):
         into `n_jobs` number of chunks.
     """
 
-    def two_neighborhood_close(G, chunk):
+    def two_neighborhood_close(adjM_filepath, chunk):
+        adjM = load(adjM_filepath, mmap_mode="r")
+        node_indices = range(adjM.shape[0])
         tnc = []
         for v in chunk:
             S = {
                 x
-                for x in G
-                if x == v or x in G[v] or any(is_path(G, [v, z, x]) for z in G)
+                for x in node_indices
+                if x == v
+                or adjM[v, x]
+                or any(adjM[v, z] and adjM[z, x] for z in node_indices)
```

Comment:
Can this be sped up with matrix multiplication? It looks like it is `adjM[v,:] @ adjM[:, x]` but only computing with indexing that are in `node_indices`.  We can limit it to only those indices using array indexing. Something like:
```python
adjM[v,node_indices] @ adjM[node_indices, x]
```